papers/subblocks.md

```diff
@@ -0,0 +1,108 @@
+first confirmation. Proposals to lower variance are introducing experimental and controversial changes in consensus protocol.
+Changing block delay via hardfork would have a lot of harsh consequences (e.g. many contracts relying on current block 
+delay would be broken). Thus it makes sense to consider weaker notions of confirmation which still could be useful for 
+a variety of applications. 
+
+Sub-Blocks
+----------
+
+A valid block is sequence of (semantically valid) header fields (and corresponding valid block sections, such as block 
+transactions), including special field to iterate over, called nonce, such as *H(b) < T*, where *H()* is Autolykos Proof-of-Work
+function, *b* are block bytes (including nonce), and *T* is a Proof-of-Work *target* value. A value which is reverse 
+to target is called difficulty *D*: *D = 2^256 / T* (in fact, slightly less value than 2^256 is taken, namely, order of 
+secp256k1 curve group, this is inherited from initial Autolykos 1 Proof-of-Work algorithm). *D* (and so *T*) is being readjusted 
+regularly via a deterministic procedure (called difficulty readjustment algorithm) to have blocks coming every two minutes on average. 
+
+Aside of blocks, *superblocks" are also used in the Ergo protocol, for building NiPoPoWs on top of them. A superblock is
+a block which is more difficult to find than an ordinary, for example, for a (level-1) superblock *S* we may require 
+*H(S) < T/2*, and in general, we can call n-level superblock a block *S* for which *H(S) < T/2^n*. Please note that a
+superblock is also a valid block (every superblock is passing block PoW test).
+
+Similarly, we can go in opposite direction and use *subblocks*, so blocks with lower difficulty. We can set *t = T/64*
+and define superblock *s* as *H(s) < t*, then miner can generate on average 64 subblocks (including normal block itself)
+per block generation period. Please note that, unlike superblocks, subblocks are not blocks, but a block is passing 
+subblock check.
+
+Subblocks are similar to block shares already used in pooled mining. Rather, this proposal is considering to use 
+sub-blocks for improving transactions propagation and providing a framework for weaker confirmations.
+
+Sub-Blocks And Transactions Propagation
+---------------------------------------
+
+Let's consider that new block is just generated. Miners A and B (among others) are working on a new block. Users are
+submitting new unconfirmed transactions at the same time to the p2p network, and eventually they are reaching miners 
+(including A and B, but at a given time a transaction could be in one of the mempools just, not necessarily both, it 
+could also be somewhere else and not known to both A and B).
+
+Then, for example, miner A is generating a sub-block committing to new transactions after last block. It sends sub-block
+header as well as weak transaction ids (6 bytes hashes) of transactions included into this sub-block but not previous 
```

Comment:
Could be worth detailing weak transaction ids a little bit, why 6 bytes of hashes specifically?